### PR TITLE
adding runtime to preparing + improving config section

### PR DESCRIPTION
**Pull request by choldgraf**
Updates `preparing` so that it includes all the possible build packs, and adds a concise list of them at the beginning of the section. This is semi-linked to https://github.com/jupyterhub/binder/pull/47, which uses `config_files.txt` to create the same section in the binder docs.

### CODE REVIEW

File: docs/source/config_files.txt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+````````````````
+::
+
+   python-2.7
+
+``Dockerfile``
+^^^^^^^^^^^^^^
+
+This will be treated as a regular Dockerfile and a regular Docker build will be performed. The presence
+of a Dockerfile will cause all other building behavior to not be triggered.
```
Comment (betatim) [inline docs/source/config_files.txt]: How about: ...of a Dockerfile prevents all other build behaviour.

**Comment by betatim** [comment]
Had a quick read and LGTM except for that small reformulation.

**Comment by choldgraf** [comment]
edit made!

**Comment by choldgraf** [comment]
anything else on this one?

**Comment by yuvipanda** [comment]
Thanks, @choldgraf!

**Pull request merged**